pkgs/shells/bash/5.1.nix

```
@@ -33,7 +33,9 @@ stdenv.mkDerivation rec {
     sha256 = "1alv68wplnfdm6mh39hm57060xgssb9vqca4yr1cyva0c342n0fc";
   };
 
-  hardeningDisable = [ "format" ];
+  hardeningDisable = [ "format" ]
+  # https://android.googlesource.com/platform/bionic/+/1342527b5791a53bf441322ab9adf41c8e060a1e%5E2..1342527b5791a53bf441322ab9adf41c8e060a1e/
+  ++ optional (stdenv.hostPlatform.useAndroidPrebuilt or false) "fortify";
```

I would do it this way; this is a property of bionic itself not just prebuilding, right?
```suggestion
  ++ optional (stdenv.hostPlatform.libc == "bionic") "fortify";
```

---

Oops, yeah we should check if libc is bionic for the hostPlatform, Bionic itself is uses "fortify" but I believe it causes bugs with other software that use fortify though (also maybe some clangisms messing this up too).

Snippet from the diff on bionic itself, although this "fix" they have doesn't work here
```
+/*
+ * Used to support clangisms with FORTIFY. Because these change how symbols are
+ * emitted, we need to ensure that bionic itself is built fortified. But lots
+ * of external code (especially stuff using configure) likes to declare
+ * functions directly, and they can't know that the overloadable attribute
+ * exists. This leads to errors like:
+ *
+ * dcigettext.c:151:7: error: redeclaration of 'getcwd' must have the 'overloadable' attribute
+ * char *getcwd ();
+ *       ^
+ *
+ * To avoid this and keep such software building, don't use overloadable if
+ * we're not using fortify.
  */
```